message.go

```diff
@@ -481,74 +485,39 @@ func (m *Message) Unmarshal(v interface{}) error {
 
 	// iterate over struct fields
 	for i := 0; i < dataStruct.NumField(); i++ {
-		fieldIndex, err := getFieldIndex(dataStruct.Type().Field(i))
-		if err != nil {
-			return fmt.Errorf("getting field %d index: %w", i, err)
-		}
-
-		// skip field without index
-		if fieldIndex < 0 {
-			continue
+		indexTag := field.NewIndexTag(dataStruct.Type().Field(i))
+		// skip field without index or if index in tag is not defined
+		if indexTag.Id < 0 {
+			return fmt.Errorf("getting field %d index error", i)
 		}
 
 		// we can get data only if field value is set
-		messageField := m.GetField(fieldIndex)
+		messageField := m.GetField(indexTag.Id)
 		if messageField == nil {
 			continue
 		}
 
-		if _, set := m.fieldsMap[fieldIndex]; !set {
+		if _, set := m.fieldsMap[indexTag.Id]; !set {
 			continue
 		}
 
 		dataField := dataStruct.Field(i)
 		switch dataField.Kind() { //nolint:exhaustive
 		case reflect.Chan, reflect.Func, reflect.Map, reflect.Pointer, reflect.UnsafePointer, reflect.Interface, reflect.Slice:
```

Comment:
This comment a bit late and is not about the scope of this ticket, but about the previous PR. Looking at this line I think we should be specific about what do we support:
```suggestion
		case reflect.Pointer, reflect.Interface, reflect.Slice:
```

the same should be for other types (Int, Int64, String)
and finally, in the `default` we should return the error.

What do you think? We can create a separate ticket for this change.